src/Agent.Listener/Configuration/NativeWindowsServiceHelper.cs

```diff
@@ -817,12 +843,513 @@ public LsaPolicy()
+                WTSWorkingDirectory,
+                WTSOEMId,
+                WTSSessionId,
+                WTSUserName,
+                WTSWinStationName,
+                WTSDomainName,
+                WTSConnectState,
+                WTSClientBuildNumber,
+                WTSClientName,
+                WTSClientDirectory,
+                WTSClientProductId,
+                WTSClientHardwareId,
+                WTSClientAddress,
+                WTSClientDisplay,
+                WTSClientProtocolType
+            }
+            public enum WTS_CONNECTSTATE_CLASS
+            {
+                WTSActive,
+                WTSConnected,
+                WTSConnectQuery,
+                WTSShadow,
+                WTSDisconnected,
+                WTSIdle,
+                WTSListen,
+                WTSReset,
+                WTSDown,
+                WTSInit
+            }
+
+            public static IntPtr OpenServer(String Name)
+            {
+                IntPtr server = WTSOpenServer(Name);
+                return server;
+            }
+
+            public static void CloseServer(IntPtr ServerHandle)
+            {
+                WTSCloseServer(ServerHandle);
+            }
+
+            public static bool IsActiveSessionExists(string testUserDomain, string testUsername)
+            {
+                var serverHandle = IntPtr.Zero;
+                serverHandle = OpenServer(Environment.MachineName);
+                var SessionInfoPtr = IntPtr.Zero;
+
+                try
+                {
+                    var userPtr = IntPtr.Zero;
+                    var domainPtr = IntPtr.Zero;
+                    var sessionCount = 0;
+                    var retVal = WTSEnumerateSessions(serverHandle, 0, 1, ref SessionInfoPtr, ref sessionCount);
+                    var dataSize = Marshal.SizeOf(typeof(WTS_SESSION_INFO));
+                    var currentSession = SessionInfoPtr;
+
+                    if (retVal != 0)
+                    {
+                        for (var i = 0; i < sessionCount; i++)
+                        {
+                            uint bytes = 0;
+                            var si = (WTS_SESSION_INFO)Marshal.PtrToStructure(currentSession, typeof(WTS_SESSION_INFO));
+                            currentSession += dataSize;
+
+                            WTSQuerySessionInformation(serverHandle, si.SessionID, WTS_INFO_CLASS.WTSUserName, out userPtr, out bytes);
+                            WTSQuerySessionInformation(serverHandle, si.SessionID, WTS_INFO_CLASS.WTSDomainName, out domainPtr, out bytes);
+
+                            var domain = Marshal.PtrToStringAnsi(domainPtr);
+                            var username = Marshal.PtrToStringAnsi(userPtr);
+
+                            Console.WriteLine("Domain : " + domain + "; Username: " + username + "; State: " + si.State);
+
+                            if (testUserDomain.Equals(domain, StringComparison.OrdinalIgnoreCase) &&
```

Comment:
ordinalIgnoreCase vs CurrentCultureIgnoreCase? would recommend Ordinal, if so use the same in other places